addons/website_sale_product_configurator/models/sale_order.py

```diff
@@ -10,8 +10,6 @@ class SaleOrder(models.Model):
     @api.multi
     def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
         lines = super(SaleOrder, self)._cart_find_product_line(product_id, line_id, **kwargs)
-        if lines:
```

Comment:
Hello @len-odoo 

First, thanks for the fix, it's indeed broken.


IMO, this part should be:

```
if line_id:
    return lines
```
instead.

When the function is passed a line_id, the post filtering part is not necessary, and even breaks the "remove from cart" feature.
eg: On your branch, add a customizable cart with optional products, then try to remove it from your cart on the "My cart" page -> it's not removed.

This would also be coherent with website_event_sale/models/sale_order.py#_cart_find_product_line

cc @JKE-be

Reply:
You are right about this. Thanks a lot for checking! I update the branch in a short while.

Reply:
It looks to be correct now; I'll see if I have time to add some test(s) before the end of the week before merging, as it seems to be a pretty essential feature of the website that is not covered.
Thanks again anyway.